### ISSUE TITLE
Support for non-Latin languages

### ISSUE THREAD

**Issue by koutheir** [iced-rs/iced#1454]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

Running the `tour` package, and typing Arabic text on the `Text input` page, displays squared interrogation point characters.

```bash
$ cargo run --package tour
```

![Screenshot](https://user-images.githubusercontent.com/1145849/193712545-14823977-9ab9-4db1-9181-9fd0165b3753.png)

### What is the expected behavior?

Arabic text should be displayed correctly.

### Version

master

### Operative System

Linux

**Comment by ids1024** [comment]
I believe with the `default_system_font` feature enabled, this should at least be able to find a font with the appropriate characters?

Though Arabic text still won't be right without RTL support (https://github.com/iced-rs/iced/issues/250) and text shaping (https://github.com/iced-rs/iced/issues/33).

**Comment by koutheir** [comment]
I added `default_system_font` to the iced features in `iced/examples/tour/Cargo.toml`:
```toml
[dependencies]
iced = { path = "../..", features = ["image", "debug", "default_system_font"] }
env_logger = "0.8"
```

and then run:
```bash
$ cargo run --package tour
```

But no Arabic characters are displayed properly.

![Screenshot](https://user-images.githubusercontent.com/1145849/193715629-a0f83af9-cba9-431c-9a2b-033b3a3c3874.png)

**Comment by Gabirel** [comment]
Same issue here on MacOS.

**Comment by adals** [comment]
> I added default_system_font to the iced features in iced/examples/tour/Cargo.toml:

<img width="801" alt="Screen Shot 2022-10-08 at 9 59 36 PM" src="https://user-images.githubusercontent.com/169691/194723385-705c0a9e-41b7-4e6d-8862-5c2588ddffc5.png">

i think that your problem is lacking a font that contains the relevant glyphs.

but RTL is still not supported yet

**Comment by koutheir** [comment]
>i think that your problem is lacking a font that contains the relevant glyphs.

I disagree, as I routinely write Arabic text in other applications (libreoffice, kate, etc.) and I don't have any font issues.

**Comment by mmstick** [comment]
@koutheir Check this out! https://github.com/pop-os/libcosmic/pull/12 Please try out Arabic in the text example and help locate bugs in the rendering.

**Comment by ids1024** [comment]
> i think that your problem is lacking a font that contains the relevant glyphs.

I think Iced isn't implementing fallback font behavior, which is normally used to find a font for a different writing system. So that's why it doesn't find a font with the right gylphs.

Luckily this (as well as text shaping and RTL) is being prototyped in libcosmic as mentioned, and hopefully will be integrated into Iced soon.

**Comment by koutheir** [comment]
> Check this out! https://github.com/pop-os/libcosmic/pull/12 Please try out Arabic in the text example and help locate bugs in the rendering.

Running the following command panics:
```bash
$ cargo run --package cosmic
<...>
     Running `target/debug/cosmic`
icon 'go-previous-symbolic' size 24 not found
icon 'window-minimize-symbolic' size 16 not found
icon 'window-maximize-symbolic' size 16 not found
icon 'window-close-symbolic' size 16 not found
icon 'network-wireless' size 20 not found
icon 'preferences-desktop' size 20 not found
icon 'system-software-update' size 20 not found
thread 'main' panicked at 'Downcast on stateless state', /home/koutheir/.cargo/git/checkouts/iced-14fbad96f5d3e3c7/7461b82/native/src/widget/tree.rs:172:28
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Running the following command works:
```bash
$ cargo run --package text
```

The Arabic letters are rendered correctly, and selecting them with the mouse and moving between the characters works as I expect.

However, using the `Right` and `Left` keyboard keys works logically, instead of visually, *i.e.*, the `Left` key moves to the right, and vice versa, and that feels weird. I don't know if that's intended or it is an issue.

**Comment by lost22git** [comment]
same profblem, any update?

**Comment by 13r0ck** [comment]
@lost22git yep! See the advanced-text branch. https://github.com/iced-rs/iced/tree/advanced-text

**Comment by spacekookie** [comment]
Arabic text renders for me in the `TextEditor` widget, but is written from left-to-right, instead of right-to-left and I don't see a way to configure how to change this

**Comment by jomaro110** [comment]
Still struggling with proper Arabic support even in 0.14.0
Hope this get resolved by next release.
